LavaQueue.cs

```diff
@@ -52,6 +52,50 @@ public bool TryDequeue(out T value)
             return true;
         }
 
+        /// <summary>
+        /// Dequeues an item from the queue at a specific index.
+        /// </summary>
+        /// <param name="index">A 0-based index</param>
+        /// <returns><see cref="T"/></returns>
+        public T RemoveAt(int index)
+        {
+            LinkedListNode<T> currentNode = _linky.First;
+            for (int i = 0; i <= index && currentNode != null; i++)
+            {
+                if (i != index)
+                {
+                    currentNode = currentNode.Next;
+                    continue;
+                }
+
+                _linky.Remove(currentNode);
+                return currentNode.Value;
+            }
+
+            throw new ArgumentOutOfRangeException("index");
```

Comment:
I feel like throwing an exception would be useless since none of the methods throw or handle exceptions in LavaQueue.